src/cmd/gray/controllers/video_transform_handler.go

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/Sogilis/Voogle/src/pkg/clients"
+	"github.com/gorilla/mux"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type VideoTransformHandler struct {
+	S3Client clients.IS3Client
+}
+
+// VideoGetMasterHandler godoc
+// @Summary Get stream video
+// @Description Get stream video
+// @Tags streams
+// @Accept plain
+// @Produce plain
+// @Param id path string true "Video ID"
+// @Success 200 {string} string "OK"
+// @Failure 400 {object} object
+// @Failure 404 {object} object
+// @Failure 500 {object} object
+// @Router /gray/transform/{videoPath} [get]
+func (v VideoTransformHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	log.Debug("GET VideoTransformHandler - parameters ", vars)
+
+	videoPath, exist := vars["videoPath"]
+	if !exist {
+		log.Error("Missing video id")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	videoPath = strings.Replace(videoPath, "+", "/", -1)
+	log.Debug("Get videoPath : ", videoPath)
+
+	object, err := v.S3Client.GetObject(context.Background(), videoPath)
+	if err != nil {
+		log.Error("Failed to open video videoPath", err)
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	log.Debug("Get video from S3 OK")
+
+	outputFile, err := transformGrayscale(object)
+	if err != nil {
+		log.Error("Cannot transformGrayscale")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	fileContent, err := os.ReadFile(outputFile)
+	if err != nil {
+		log.Error("Unable read output file", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	readerFile := bytes.NewReader(fileContent)
+	if _, err := io.Copy(w, readerFile); err != nil {
+		log.Error("Unable to stream subpart", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
+
+func transformGrayscale(video io.Reader) (string, error) {
+	buf, err := io.ReadAll(video)
+	if err != nil {
+		return "", err
+	}
+	inputFilename := "input.ts"
+	f, err := os.Create(inputFilename)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = f.Write(buf)
+	if err != nil {
+		return "", err
+	}
+
+	f.Close()
+
+	outputFilename := "output.ts"
+	// cmd := exec.Command("ffmpeg", "-y", "-i", inputFilename, "-vf", "format=gray", "-c:a", "copy", outputFilename)
+	cmd := exec.Command("ffmpeg", "-y", "-i", inputFilename, "-vf", "vflip", "-c:a", "copy", outputFilename)
```

Comment:
We can regroup all the different processing of videos in a package and not in a handler.